modules/protocol/src/vector.ts

```diff
@@ -332,6 +393,97 @@ export class Vector implements IVectorProtocol {
     return undefined;
   }
 
+  // Adds a given task to the internal queue
+  // TODO: implement
+  private addToQueue(
+    params: UpdateParams<any>,
+    previous?: ChannelUpdate,
+  ): Promise<
+    Result<
+      {
+        updatedChannel: FullChannelState;
+        updatedActiveTransfers?: FullTransferState[];
+        updatedTransfer?: FullTransferState;
+      },
+      QueuedUpdateError
+    >
+  > {
+    throw new Error("addToQueue method not implemented");
+  }
+
+  // Adds a given task to the front of the internal queue
+  // TODO: implement
+  private addToFrontOfQueue(
+    params: UpdateParams<any>,
+    previous?: ChannelUpdate,
+  ): Promise<
+    Result<
+      {
+        updatedChannel: FullChannelState;
+        updatedActiveTransfers?: FullTransferState[];
+        updatedTransfer?: FullTransferState;
+      },
+      QueuedUpdateError
+    >
+  > {
+    throw new Error("addToQueue method not implemented");
+  }
+
+  /**
+   * Returns true if the received upddate includes our proposed update. If true,
+   * this means you don't have to re-send your proposed update. If false, this
+   * means you should resend your proposed upddate
+   * @param receivedUpdate The update you have gotten from your counterparty
+   * @param proposedUpdate The update you have sent out to counterparty
+   */
+  private includesOurProposedUpdate(
+    receivedUpdate: ChannelUpdate,
+    proposedUpdate: ChannelUpdate,
+    // TODO: should probably just use the generated merkle tree here instead
+    // of regenerating it from the transfers and channel
+    updatedActiveTransfers: FullTransferState[],
+    proposedTransfer?: FullTransferState,
+  ): Result<boolean, QueuedUpdateError> {
+    // If both are a setup update, your update would fail and the ultimate
+    // result (a channel is set up) is achieved
+    if (receivedUpdate.type === UpdateType.setup && proposedUpdate.type === UpdateType.setup) {
+      return Result.ok(true);
+    }
+
+    // If both are a deposit update, your deposit is implicitly included
+    if (receivedUpdate.type === UpdateType.deposit && proposedUpdate.type === UpdateType.deposit) {
+      return Result.ok(true);
+    }
+
+    // If both are a resolve, it would *not* include your proposed update since
+    // only the responder to a transfer can call resolve. Updates by definition
+    // are different
```

Comment:
I don't follow here.

The reason that the proposed update could include your resolve is that you just sent them the resolve.

The thing we're checking for here generally is that they received your update, processed it, sent a response, then processed their update and sent an update but for whatever reason the first response was dropped or is received out of order.

In the scenario above it would include your proposed update.